### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by PadowYT2**
* define -> const
* Remove unreachable statements
* Fix page time load
* Missing property's type declarations
* Inline variables
* Remove redundant arguments
* Replace deprecated stuff
* Fix footer on index page
* Formatting for Smarty files
* Remove Default template (it is pr13 come on, not pr8)

Also, dm Derkades about [translate.namelessmc.com](https://translate.namelessmc.com)
https://cdn.padow.ru/VmWp

### CODE REVIEW

File: upload/custom/panel_templates/Default/craftingstore/index.tpl [hunk 1]
```diff
@@ -2,84 +2,91 @@
...
-    <!-- End Wrapper -->
-</div>
-
-{include file='scripts.tpl'}
+    {include file='scripts.tpl'}
 
 </body>
-</html>
+
+</html>
```
Comment (Tim-NL) [inline upload/custom/panel_templates/Default/craftingstore/index.tpl:92]: Could you add a newline to the end?

File: upload/custom/panel_templates/Default/craftingstore/sync.tpl [hunk 2]
```diff
@@ -65,24 +65,25 @@
...
-    <!-- End Wrapper -->
-</div>
-
-{include file='scripts.tpl'}
+    {include file='scripts.tpl'}
 
 </body>
-</html>
+
+</html>
```
Comment (Tim-NL) [inline upload/custom/panel_templates/Default/craftingstore/sync.tpl:89]: Could you add a newline to the end?

File: upload/modules/CraftingStore/module.php [hunk 3]
```diff
@@ -5,37 +5,37 @@ class CraftingStoreModule extends Module
     /**
      * @var InstallDatabaseFlow
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/module.php:6]: This comment is obsolete if you add a type, could you remove those docs for the items?

File: upload/modules/CraftingStore/module.php [hunk 4]
```diff
@@ -71,16 +71,7 @@ public function __construct(
...
-     * @var User $user
-     * @var Pages $pages
-     * @var Cache $cache
-     * @var Smarty $smarty
-     * @var Navigation[] $navs
-     * @var Widget $widgets
-     * @var null $template
-     */
-    public function onPageLoad($user, $pages, $cache, $smarty, $navs, $widgets, $template)
+    public function onPageLoad(User $user, Pages $pages, Cache $cache, Smarty $smarty, $navs, Widgets $widgets, $template)
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/module.php:74]: Some of the items in this list do not have types, and the doc has been removed. For methods that have items that cannot have a (correct) typehint (e.g. an array filled with a specific object) I would like to still have the normal doc-block so the idea knows what to do with it.

I also see that some do not have a typehint, why is that?

File: upload/modules/CraftingStore/src/Category/Builder/CategoryBuilder.php [hunk 5]
```diff
@@ -5,12 +5,12 @@ class CategoryBuilder
     /**
      * @var CategoryRepository
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Category/Builder/CategoryBuilder.php:6]: These docblocks are no longer required with this change.

File: upload/modules/CraftingStore/src/Database/Repository/CategoryRepository.php [hunk 6]
```diff
@@ -2,30 +2,25 @@
...
     public function truncate(): void
     {
-        $this->db->createQuery('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::CATEGORIES);
+        $this->db->query('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::CATEGORIES);
     }
 
     public function create(int $id, int $order, string $name, ?string $description, ?int $parentCategory = null): void
     {
-        $this->queries->create(DatabaseTableEnum::CATEGORIES, [
+        DB::getInstance()->insert(DatabaseTableEnum::CATEGORIES, [
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Database/Repository/CategoryRepository.php:23]: Could you please use the injected instance instead? (same for the ones below)

File: upload/modules/CraftingStore/src/Database/Repository/PackageRepository.php [hunk 7]
```diff
@@ -2,30 +2,25 @@
...
     public function truncate(): void
     {
-        $this->db->createQuery('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::PACKAGES);
+        $this->db->query('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::PACKAGES);
     }
 
     public function create(int $id, int $categoryId, int $order, string $name, string $price, string $description, ?string $image): void
     {
-        $this->queries->create(DatabaseTableEnum::PACKAGES, [
+        DB::getInstance()->insert(DatabaseTableEnum::PACKAGES, [
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Database/Repository/PackageRepository.php:23]: Could you please use the injected db instead?

File: upload/modules/CraftingStore/src/Database/Repository/PaymentRepository.php [hunk 8]
```diff
@@ -2,30 +2,25 @@
 
 class PaymentRepository
 {
-    /**
-     * @var Queries
-     */
-    protected $queries;
 
     /**
      * @var DB
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Database/Repository/PaymentRepository.php:7]: This block is no longer required.

File: upload/modules/CraftingStore/src/Database/Repository/PaymentRepository.php [hunk 9]
```diff
@@ -2,30 +2,25 @@
...
     public function truncate(): void
     {
-        $this->db->createQuery('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::PAYMENTS);
+        $this->db->query('TRUNCATE TABLE ' . DatabaseTableEnum::PREFIX . DatabaseTableEnum::PAYMENTS);
     }
 
     public function create(int $price, int $timestamp, ?string $playerId, string $playerName, string $packageName): void
     {
-        $this->queries->create(DatabaseTableEnum::PAYMENTS, [
+        DB::getInstance()->insert(DatabaseTableEnum::PAYMENTS, [
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Database/Repository/PaymentRepository.php:23]: Could you please use the injected db instead?

File: upload/modules/CraftingStore/src/Database/Repository/SettingRepository.php [hunk 10]
```diff
@@ -2,19 +2,10 @@
 
 class SettingRepository
 {
-    /**
-     * @var Queries
-     */
-    protected $queries;
-
-    public function __construct(Queries $queries)
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Database/Repository/SettingRepository.php:10]: Could in inject Db here, and use it in the methods below?

File: upload/modules/CraftingStore/src/Install/Flow/InstallDatabaseFlow.php [hunk 11]
```diff
@@ -2,66 +2,41 @@
 
 class InstallDatabaseFlow
 {
-    /**
-     * @var Queries
-     */
-    protected $queries;
-
-    public function __construct(Queries $queries)
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Install/Flow/InstallDatabaseFlow.php:10]: Could you inject the Db, and use that instead in this class?

File: upload/modules/CraftingStore/src/Install/Flow/UpdatePermissionFlow.php [hunk 12]
```diff
@@ -2,29 +2,19 @@
 
 class UpdatePermissionFlow
 {
-    /**
-     * @var Queries
-     */
-    protected $queries;
-
-    public function __construct(Queries $queries)
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Install/Flow/UpdatePermissionFlow.php:10]: Could you inject Db and use that instead in this class?

File: upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php [hunk 13]
```diff
@@ -5,12 +5,12 @@ class BackendNavigationBuilder
     /**
      * @var NavigationOrderRetriever
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php:6]: Docblocks are no longer required.

File: upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php [hunk 14]
```diff
@@ -25,7 +25,7 @@ public function build(Navigation $navigation, User $user): void
                 if ($user->hasPermission(PermissionEnum::SETTINGS)) {
                     $order = $this->navigationOrderRetriever->retrieve();
 
-                    $navigation->add('craftingstore_divider', mb_strtoupper($this->craftingStoreLanguage->get(LanguageEnum::PREFIX, LanguageEnum::CRAFTING_STORE)), 'divider', 'top', null, $order, '');
+                    $navigation->add('craftingstore_divider', mb_strtoupper($this->craftingStoreLanguage->get(LanguageEnum::PREFIX, LanguageEnum::CRAFTING_STORE)), 'divider', 'top', null, $order);
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php:28]: Why did you change this line?

File: upload/modules/CraftingStore/src/Navigation/Builder/UserNavigationBuilder.php [hunk 15]
```diff
@@ -5,17 +5,17 @@ class UserNavigationBuilder
     /**
      * @var NavigationOrderRetriever
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Navigation/Builder/UserNavigationBuilder.php:6]: Docblocks are no longer required.

File: upload/modules/CraftingStore/src/Navigation/Retriever/NavigationOrderRetriever.php [hunk 16]
```diff
@@ -5,7 +5,7 @@ class NavigationOrderRetriever
     /**
      * @var Cache
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Navigation/Retriever/NavigationOrderRetriever.php:6]: Docblock is no longer required.

File: upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php [hunk 17]
```diff
@@ -5,27 +5,32 @@ class FullSyncFlow
     /**
      * @var InformationRetriever
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php:6]: Docblocks are no longer required.

File: upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php [hunk 18]
```diff
@@ -45,7 +50,7 @@ public function __construct(
 
     public function performFlow(): bool
     {
-        $serverKey = $this->settingRepository->firstValueByName(SettingEnum::SERVER_KEY, null);
+        $serverKey = $this->settingRepository->firstValueByName(SettingEnum::SERVER_KEY);
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php:53]: Why did you remove the last value? I understand that the default value for that param is `null`, but I'd like to be as explicit/strict as possible.

File: upload/modules/CraftingStore/src/Sync/Flow/UpdateCategoryFlow.php [hunk 19]
```diff
@@ -5,12 +5,12 @@ class UpdateCategoryFlow
     /**
      * @var CategoryUpdater
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Flow/UpdateCategoryFlow.php:6]: Blocks no longer required.

File: upload/modules/CraftingStore/src/Sync/Flow/UpdatePackageFlow.php [hunk 20]
```diff
@@ -5,12 +5,12 @@ class UpdatePackageFlow
     /**
      * @var PackageUpdater
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Flow/UpdatePackageFlow.php:6]: Blocks no longer required.

File: upload/modules/CraftingStore/src/Sync/Flow/UpdatePaymentFlow.php [hunk 21]
```diff
@@ -5,12 +5,12 @@ class UpdatePaymentFlow
     /**
      * @var PaymentUpdater
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Flow/UpdatePaymentFlow.php:6]: Blocks no longer required.

File: upload/modules/CraftingStore/src/Sync/Updater/CategoryUpdater.php [hunk 22]
```diff
@@ -5,7 +5,7 @@ class CategoryUpdater
     /**
      * @var CategoryRepository
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Updater/CategoryUpdater.php:6]: Block no longer required.

File: upload/modules/CraftingStore/src/Sync/Updater/InformationUpdater.php [hunk 23]
```diff
@@ -5,7 +5,7 @@ class InformationUpdater
     /**
      * @var SettingRepository
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Updater/InformationUpdater.php:6]: Block no longer required.

File: upload/modules/CraftingStore/src/Sync/Updater/PackageUpdater.php [hunk 24]
```diff
@@ -5,7 +5,7 @@ class PackageUpdater
     /**
      * @var PackageRepository
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Updater/PackageUpdater.php:6]: Block no longer required.

File: upload/modules/CraftingStore/src/Sync/Updater/PaymentUpdater.php [hunk 25]
```diff
@@ -5,7 +5,7 @@ class PaymentUpdater
     /**
      * @var PaymentRepository
```
Comment (Tim-NL) [inline upload/modules/CraftingStore/src/Sync/Updater/PaymentUpdater.php:6]: Block no longer required.

**Review by Tim-NL — CHANGES_REQUESTED** [review]

**Comment by Tim-NL** [comment]
Thanks!

I added some comments (mostly duplicated commands for the same thing in multiple classes).

And I noticed that you changed the permission on some of the files to `755`, why did you do that exactly (you can let git ignore permission changes with a quick command for this specific project). I don't think that permission `755` is the best permission for files.

-Tim

**Comment by PadowYT2** [comment]
> Thanks!
> 
> I added some comments (mostly duplicated commands for the same thing in multiple classes).
> 
> And I noticed that you changed the permission on some of the files to `755`, why did you do that exactly (you can let git ignore permission changes with a quick command for this specific project). I don't think that permission `755` is the best permission for files.
> 
> -Tim

1. Yep, I'll do them
2. When I needed it for show on the NamelessMC website I needed them to be `777`, and uploading to Github it is `755`, might show the commands for making the files right? (I've used `chmod -R 755 .`)

File: upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php (hunk 14 above)
Comment (PadowYT2) [reply upload/modules/CraftingStore/src/Navigation/Builder/BackendNavigationBuilder.php:28]: As PHPStorm suggests `Remove redundant arguments`

File: upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php (hunk 18 above)
Comment (PadowYT2) [reply upload/modules/CraftingStore/src/Sync/Flow/FullSyncFlow.php:53]: As PHPStorm suggests `Remove redundant arguments`

**Comment by Tim-NL** [comment]
> > Thanks!
> > I added some comments (mostly duplicated commands for the same thing in multiple classes).
> > And I noticed that you changed the permission on some of the files to `755`, why did you do that exactly (you can let git ignore permission changes with a quick command for this specific project). I don't think that permission `755` is the best permission for files.
> > -Tim
> 
> 1. Yep, I'll do them
> 2. When I needed it for show on the NamelessMC website I needed them to be `777`, and uploading to Github it is `755`, might show the commands for making the files right? (I've used `chmod -R 755 .`)

You can run `git config core.fileMode false` in your project folder, so git ignores any permission changes. But before you do that, please undo those permission changes. 

You should generally never need to give files permission `755` (making them world readable / executable). For the files of our module it is not really an issue, but if you do this to every file (e.g. the config file of Nameless) then any user on the same webserver can view your database/mail credentials. If the files & folders belong to the right user, and your PHP FPM (if you use Nginx) are configured to use the same user (e.g. a specific user for this website) then you should never need to change the file permissions. And if one of the websites on this machine is breached, it does not allow access to the others - Like it should. 

Giving files `777` means that they are world readable/executable and editable. If you have to do that to make it work, then your webserver or the user that the files belong to is not configured correctly. You should never need to do that.

-Tim
